lib/EmitVerilog/EmitVerilog.cpp

```diff
@@ -1950,6 +1953,47 @@ void ModuleEmitter::emitDecl(MemOp op) {
   }
 }
 
+void ModuleEmitter::emitDecl(sv::InterfaceOp op) {
+  os << "interface " << op.sym_name() << ";\n";
+
+  addIndent();
+  for (auto &op : op.getBodyBlock()->without_terminator())
+    emitOperation(&op);
+  reduceIndent();
+
+  os << "endinterface\n\n";
+}
+
+void ModuleEmitter::emitDecl(sv::InterfaceSignalOp op) {
+  indent() << "logic ";
+
+  auto type = op.type();
+  if (getBitWidthOrSentinel(type) != 1) {
+    emitTypePaddedToWidth(type, 0, op);
+    os << ' ';
+  }
+
+  os << op.sym_name() << ";\n";
+}
+
+void ModuleEmitter::emitDecl(sv::InterfaceModportOp op) {
+  indent() << "modport " << op.sym_name() << '(';
+  ArrayAttr portsAttr = op.ports();
+
+  for (size_t i = 0, e = portsAttr.size(); i != e; ++i) {
```

Comment:
I'd recommend using the llvm::interleaveComma algorithm in ADT/STLExtras.h

Reply:
You can pass a lambda to interleaveComma like this: https://github.com/llvm/llvm-project/blob/d69762c404dedb8af915c503e0ac40c8e57ddf84/mlir/tools/mlir-tblgen/TypeDefGen.cpp#L100